Issue #3067282 by volkerk, chr.fritsch, andrew.martin, BetoAveiga, quicksketch: Autocreate values are lost on node preview and autosave resume

ol0lll opened the pull request.

ol0lll committed:
1st

ol0lll committed:
Merge branch '8.x-1.x' into fix/losing-autocreate-values

chrfritsch reviewed (changes requested):
- Doesn't work without autocomplete
- We need tests
- Please move the other fix with a test in a separate issue

ol0lll committed:
Revert changes in getValidReferenceableEntities() see http://drupal.org/i/3068450, added auto_create handling.

ol0lll commented:
* Moved getValidReferenceableEntities() to https://github.com/thunder/select2/pull/66
* Added handling for widget without autocomplete but with auto_create

I would prefer another review before writing tests.

chrfritsch committed:
Merge branch '8.x-1.x' into fix/losing-autocreate-values

chrfritsch commented:
Scenario 1:
- Autocreate: On
- Cardinality: Unlimited
- Autocomplete: Off 

The select box only shows already selected entities. Not all existing ones.

Scenario 2:
- Autocreate: On
- Cardinality: 1
- Autocomplete: On 

A newly created entity is gone after the return from the preview

chrfritsch committed:
Refactor tests

chrfritsch committed:
More tests

ol0lll committed:
move stuff and revert changes

ol0lll committed:
Merge branch 'fix/losing-autocreate-values' of github.com:thunder/select2 into fix/losing-autocreate-values

ol0lll committed:
Merge branch '8.x-1.x' of github.com:thunder/select2 into fix/losing-autocreate-values

ol0lll committed:
add groupd

ol0lll committed:
Add handling for single value field

chrfritsch committed:
Preview tests

ol0lll committed:
wait for highlighted options

chrfritsch committed:
Add more tests

chrfritsch committed:
Use assertEquals

chrfritsch committed:
Merge branch '8.x-1.x' into fix/losing-autocreate-values
